src/lib/chawathe98_matcher.rs

```diff
@@ -2086,16 +2066,18 @@ impl<T: Clone + Debug + Eq + 'static> MatchingTreesScriptor<T> for Config2 {
 
             // Lets get edge E2
             let e2_to_node_checker = helper_function.1.get(&m);
-            let mut e2_to_node = None;
-            if e2_to_node_checker.is_some() {
-                e2_to_node = e2_to_node_checker.unwrap().0.iter().nth(0);
-            }
+            let e2_to_node = if e2_to_node_checker.is_some() {
```

Comment:
Could this be simplified to something like `let e2_to_node = e2_to_node_checker.map(|x| x.0.first());`?